resolvers/node/resolution.rs

```diff
@@ -63,6 +66,30 @@ fn conditions_from_resolution_mode(
   }
 }
 
+pub struct ConditionsFromResolutionMode {
```

Comment:
Maybe this should just be: `pub struct ConditionsFromResolutionMode(Option<Box<dyn Fn(ResolutionMode) -> &'static [&'static str]>>)` with a default implementation of `ConditionsFromResolutionMode(None)` that will instead use `deno_conditions_from_resolution_mode`? That way the other code can just do `ConditionsFromResolutionMode::default()` instead of needing to know about `deno_conditions_from_resolution_mode`